lang/java/reef-checkpoint/src/main/java/org/apache/reef/io/checkpoint/fs/FSCheckpointService.java

```diff
@@ -75,78 +75,90 @@ static final Path tmpfile(final Path p) {
   public CheckpointWriteChannel create()
       throws IOException {
 
-    final String name = namingPolicy.getNewName();
-
+    final String name = namingPolicy.getNewCheckpointName();
     final Path p = new Path(name);
     if (p.isUriPathAbsolute()) {
-      throw new IOException("Checkpoint cannot be an absolute path");
+      throw new IOException("Checkpoint name cannot be an absolute path.");
     }
-    return createInternal(new Path(base, p));
+
+    return createInternal(new Path(basePath, p));
   }
 
   CheckpointWriteChannel createInternal(final Path name) throws IOException {
-
     //create a temp file, fail if file exists
```

Comment:
It might be necessary because there is apparently no reliable way no check file permissions in advance.  This implementation is built around Hadoop `FileSystem` class, which has `access` method for checking permissions, but the Javadoc for this method says:

> In general, applications should avoid using this method, due to the risk of
    time-of-check/time-of-use race conditions.  The permissions on a file may
   change immediately after the access call returns.  Most applications should
   prefer running specific file system actions as the desired user represented
    by a  UserGroupInformation.